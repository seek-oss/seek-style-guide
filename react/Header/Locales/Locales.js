import styles from './Locales.less';
import React from 'react';
import PropTypes from 'prop-types';

import classnames from 'classnames';
import ScreenReaderOnly from '../../ScreenReaderOnly/ScreenReaderOnly';

export default function Locales({ locale, linkRenderer }) {
  const isAU = locale === 'AU';
  const isNZ = locale === 'NZ';

  const listClasses = classnames({
    [styles.list]: true,
    [styles.list_isNZ]: isNZ
  });

  return (
    <nav aria-labelledby="Locales" role="navigation" className={styles.root}>

      <ScreenReaderOnly>
        <h1 id="Locales">Select your country</h1>
      </ScreenReaderOnly>

      <ul className={listClasses}>
        <li className={styles.listItem}>
          {
            isAU ?
              <span className={styles.locale_isActive}>AU</span> :
              linkRenderer({
                'data-analytics': 'header:au+homepage',
                className: styles.localeLink,
                href: 'https://www.seek.com.au',
                title: 'SEEK Australia',
                children: 'AU'
              })
          }
        </li>
        <li className={styles.listItem}>
          {
            isNZ ?
              <span className={styles.locale_isActive}>NZ</span> :
              linkRenderer({
                'data-analytics': 'header:nz+homepage',
                className: styles.localeLink,
                href: 'https://www.seek.co.nz',
                title: 'SEEK New Zealand',
                children: 'NZ'
              })
          }
        </li>
      </ul>

    </nav>
  );
}

Locales.propTypes = {
  locale: PropTypes.string.isRequired,
  linkRenderer: PropTypes.func.isRequired
};
