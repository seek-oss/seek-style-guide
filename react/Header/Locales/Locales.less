@import (reference) "~seek-style-guide/theme";

.list {
  display: flex;
}

.list_isNZ {
  flex-direction: row-reverse;
}

.locale_isActive {
  .rawText();
  height: (@grid-row-height * 3);
  display: flex;
  align-items: center;
  box-shadow: inset 0 -2px @sk-black;
  padding: 0;
  font-weight: @sk-bold;
  color: @sk-black;
  margin-right: 15px;

  &:hover {
    color: @sk-black;
    cursor: auto;
  }
}

.localeLink {
  .nav_link();
  .rawText();
  height: (@grid-row-height * 3);
  display: flex;
  align-items: center;
  color: @sk-mid-gray-dark;

  &:hover {
    color: @sk-black;
  }
}
