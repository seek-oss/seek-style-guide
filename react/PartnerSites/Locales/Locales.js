import styles from './Locales.less';
import React from 'react';
import PropTypes from 'prop-types';

import classnames from 'classnames';

export default function Locales({ locale, linkRenderer, localeLinks }) {
  const isAU = locale === 'AU';
  const isNZ = locale === 'NZ';

  const listClasses = classnames({
    [styles.list]: true,
    [styles.list_isNZ]: isNZ
  });

  return (
    <nav aria-label="country" role="navigation" className={styles.root}>
      <ul className={listClasses}>
        <li className={styles.listItem}>
          {isAU ? (
            <span className={styles.locale_isActive}>AU</span>
          ) : (
            linkRenderer({
              className: styles.localeLink,
              children: 'AU',
              ...localeLinks.AU
            })
          )}
        </li>
        <li className={styles.listItem}>
          {isNZ ? (
            <span className={styles.locale_isActive}>NZ</span>
          ) : (
            linkRenderer({
              className: styles.localeLink,
              children: 'NZ',
              ...localeLinks.NZ
            })
          )}
        </li>
      </ul>
    </nav>
  );
}

const localeLink = PropTypes.shape({
  'data-analytics': PropTypes.string,
  href: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
});

Locales.propTypes = {
  locale: PropTypes.string.isRequired,
  linkRenderer: PropTypes.func.isRequired,
  localeLinks: PropTypes.shape({
    AU: localeLink.isRequired,
    NZ: localeLink.isRequired
  }).isRequired
};
