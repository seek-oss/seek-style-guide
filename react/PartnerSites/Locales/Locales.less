@import (reference) "~seek-style-guide/theme";

.list {
  display: flex;
}

.list_isNZ {
  flex-direction: row-reverse;
}

.locale_isActive {
  .rawText(@small-type-scale);
  height: (@row-height * 5);
  display: flex;
  align-items: center;
  box-shadow: inset 0 -2px @sk-black;
  padding: 0;
  font-weight: @sk-bold;
  color: @sk-black;
  margin-right: 15px;

  &:hover {
    color: @sk-black;
    cursor: auto;
  }
}

.localeLink {
  .nav_link();
  .rawText(@small-type-scale);
  height: (@row-height * 5);
  display: flex;
  align-items: center;
  color: @sk-mid-gray-dark;

  &:hover {
    color: @sk-black;
  }
}
