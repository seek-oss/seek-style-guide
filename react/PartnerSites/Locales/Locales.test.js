import React from 'react';
import { render } from 'enzyme';

import Locales from './Locales';

const renderLocales = props => render(<Locales {...props} />);
const linkRenderer = props => <a {...props} />;
const localeLinks = {
  AU: {
    'data-analytics': 'header:au+homepage',
    href: 'https://www.seek.com.au',
    title: 'SEEK Australia',
    children: 'AU'
  },
  NZ: {
    'data-analytics': 'header:nz+homepage',
    href: 'https://www.seek.co.nz',
    title: 'SEEK New Zealand',
    children: 'NZ'
  }
};

describe('Locales:', () => {
  it('should render with locale of AU first and active', () => {
    expect(
      renderLocales({ locale: 'AU', linkRenderer, localeLinks })
    ).toMatchSnapshot();
  });

  it('should render with locale of NZ first and active', () => {
    expect(
      renderLocales({ locale: 'NZ', linkRenderer, localeLinks })
    ).toMatchSnapshot();
  });
});
