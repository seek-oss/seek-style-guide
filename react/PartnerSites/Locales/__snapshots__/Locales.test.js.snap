// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Locales: should render with locale of AU first and active 1`] = `
<nav
  aria-labelledby="Locales"
  role="navigation"
>
  <span
    class="ScreenReaderOnly__root"
  >
    <h1
      id="Locales"
    >
      Select your country
    </h1>
  </span>
  <ul
    class="Locales__list"
  >
    <li>
      <span
        class="Locales__locale_isActive"
      >
        AU
      </span>
    </li>
    <li>
      <a
        class="Locales__localeLink"
        data-analytics="header:nz+homepage"
        href="https://www.seek.co.nz"
        title="SEEK New Zealand"
      >
        NZ
      </a>
    </li>
  </ul>
</nav>
`;

exports[`Locales: should render with locale of NZ first and active 1`] = `
<nav
  aria-labelledby="Locales"
  role="navigation"
>
  <span
    class="ScreenReaderOnly__root"
  >
    <h1
      id="Locales"
    >
      Select your country
    </h1>
  </span>
  <ul
    class="Locales__list Locales__list_isNZ"
  >
    <li>
      <a
        class="Locales__localeLink"
        data-analytics="header:au+homepage"
        href="https://www.seek.com.au"
        title="SEEK Australia"
      >
        AU
      </a>
    </li>
    <li>
      <span
        class="Locales__locale_isActive"
      >
        NZ
      </span>
    </li>
  </ul>
</nav>
`;
